[Fleet] Exclude events log from EA diagnostic collection in modal

lucabelluccini opened the pull request:
## Summary

In 8.15 we've reintroduced the logging of raw events when in debug log via https://github.com/elastic/beats/pull/38767.

Given the Elastic Agent is able to read the flag from the action (https://github.com/elastic/elastic-agent/blob/44528c49507c18e71a4cfd2a6ba4f0904bd32009/internal/pkg/fleetapi/action.go#L483), I'm attempting to draft a PR.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### Risk Matrix

- Is it safe to send the new parameter to Elastic Agents which do not support the `exclude_event_logs`?
- Do we need to handle the saved objects types of the Diagnostic Request?

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

lucabelluccini committed:
Initial commit for exclude events log from EA diagnostic

lucabelluccini committed:
Identify tests to add + add the parameter to the openapi spec

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/bundled.json:
```diff
@@ -3205,6 +3205,9 @@
                       ]
                     }
+                  },
+                  "exclude_event_logs": {
```
```suggestion
                  "exclude_events_log": {
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/bundled.json

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/bundled.json:
```diff
@@ -3307,6 +3310,9 @@
                       ]
                     }
+                  },
+                  "exclude_event_logs": {
```
```suggestion
                  "exclude_events_log": {
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/bundled.json

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/bundled.yaml:
```diff
@@ -2014,6 +2014,9 @@ paths:
                       - type: string
                         enum:
                           - CPU
+                exclude_event_logs:
```
```suggestion
                exclude_events_log:
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/bundled.yaml

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/bundled.yaml:
```diff
@@ -2072,6 +2075,9 @@ paths:
                       - type: string
                         enum:
                           - CPU
+                exclude_event_logs:
```
```suggestion
                exclude_events_log:
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/bundled.yaml

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/paths/agents@bulk_request_diagnostics.yaml:
```diff
@@ -40,6 +40,8 @@ post:
                   - type: string
                     enum:
                       - "CPU"
+            exclude_event_logs:
```
```suggestion
            exclude_events_log:
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/paths/agents@bulk_request_diagnostics.yaml

lucabelluccini commented on x-pack/plugins/fleet/common/openapi/paths/agents@{agent_id}@request_diagnostics.yaml:
```diff
@@ -21,6 +21,8 @@ post:
                   - type: string
                     enum:
                       - "CPU"
+            exclude_event_logs:
```
```suggestion
            exclude_events_log:
```

lucabelluccini committed:
Update x-pack/plugins/fleet/common/openapi/paths/agents@{agent_id}@request_diagnostics.yaml

lucabelluccini commented on x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_request_diagnostics_modal/index.tsx:
```diff
@@ -37,19 +38,23 @@ export const AgentRequestDiagnosticsModal: React.FunctionComponent<Props> = ({
     try {
       setIsSubmitting(true);
       const additionalMetrics = cpuMetricsEnabled ? [RequestDiagnosticsAdditionalMetrics.CPU] : [];
+      const excludeEventsLog = includeEventsLogEnabled ? false : true;
```
```suggestion
      const excludeEventsLog = !includeEventsLogEnabled;
```

lucabelluccini committed:
Update x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_request_diagnostics_modal/index.tsx

lucabelluccini committed:
Update / Add tests for exclude event data

lucabelluccini committed:
Fix linting

lucabelluccini commented:
Tested locally with 8.15 and 8.11 to ensure the flag is not preventing the diagnostic collection on older Elastic Agent versions.

We might need to put this on hold until Elastic Agent considers this flag in the protocol and repeat the tests as afaik this requires a protocol change.

lucabelluccini commented on x-pack/plugins/fleet/server/services/agents/request_diagnostics.ts:
```diff
@@ -23,7 +23,8 @@ import {
   agentId: string,
-  additionalMetrics?: RequestDiagnosticsAdditionalMetrics[]
+  additionalMetrics?: RequestDiagnosticsAdditionalMetrics[],
+  excludeEventsLog?: boolean,
```
Not 100% sure

lucabelluccini commented on x-pack/plugins/fleet/server/routes/agent/request_diagnostics_handler.test.ts:
```diff
@@ -36,7 +36,7 @@ describe('request diagnostics handler', () => {
     { agentId: string },
     undefined,
-    { additional_metrics: RequestDiagnosticsAdditionalMetrics[] },
+    { additional_metrics: RequestDiagnosticsAdditionalMetrics[]; exclude_events_log?: boolean },
```
Not 100% sure

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

nchaulet commented:
Got the following error when trying the API

<img width="1174" alt="Screenshot 2024-08-26 at 9 27 51 AM" src="https://github.com/user-attachments/assets/b6ca4970-37bd-401c-a343-080d99735565">

You will have to update the request schema for the POST request diagnostic and bulk operation too [here](https://github.com/lucabelluccini/kibana/blob/c1fbd970934614e831752c77c7c1daedd5638938/x-pack/plugins/fleet/server/types/rest_spec/agent.ts#L171)

nchaulet commented on x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_request_diagnostics_modal/index.tsx:
```diff
@@ -128,15 +132,20 @@ export const AgentRequestDiagnosticsModal: React.FunctionComponent<Props> = ({
+      <EuiCheckbox
+        id="includeEventsLogCheckbox"
+        data-test-subj="includeEventsLogCheckbox"
+        label="Include Events Logs (might contain sensible information)"
```
It will be nice to have translated value with `i18n.translate` for the label

lucabelluccini committed:
Move checkbox labels to i18n

lucabelluccini commented:
Weird to get the warning - I do not get the popup. I've already updated the request schema.

I've updated the code to use the i18n and the render is still good.

<img width="815" alt="image" src="https://github.com/user-attachments/assets/7562ca84-a190-4bc4-9021-6ef579f520e6">

The derived Fleet actions SO are:
```
     {
        "_index": ".fleet-actions-7",
        "_id": "6d9e72e1-504c-4b3f-bad4-3b7791282597",
        "_score": null,
        "_source": {
          "@timestamp": "2024-08-26T21:52:59.529Z",
          "expiration": "2024-08-27T00:52:59.529Z",
          "agents": [
            "a04e9ed5-eccb-43ea-a26b-8d0ee7010ba7"
          ],
          "action_id": "817f60c0-2af5-4b60-8ccf-be58c4601ed4",
          "data": {
            "additional_metrics": [],
            "exclude_events_log": false
          },
          "type": "REQUEST_DIAGNOSTICS",
          "traceparent": "00-cb510e4d60758d79731b1678971bce38-25ae78c9627e0a56-01"
        },
        "sort": [
          1724709179529
        ]
      },
      {
        "_index": ".fleet-actions-7",
        "_id": "fdc86319-ad15-4d2e-8f19-0e719bd46f66",
        "_score": null,
        "_source": {
          "@timestamp": "2024-08-26T21:52:50.542Z",
          "expiration": "2024-08-27T00:52:50.542Z",
          "agents": [
            "a04e9ed5-eccb-43ea-a26b-8d0ee7010ba7"
          ],
          "action_id": "22c2a9f4-71ae-4d43-a3d0-8641fb7ffef0",
          "data": {
            "additional_metrics": [],
            "exclude_events_log": true
          },
          "type": "REQUEST_DIAGNOSTICS",
          "traceparent": "00-6a1b2c5a804f01dc2a2bd30c314a8033-66ccb911e742654b-01"
        },
        "sort": [
          1724709170542
        ]
      }
```

I still think we need some work on EA side in order to merge IIUC.

jlind23 commented:
@lucabelluccini just created [this](https://github.com/elastic/elastic-agent/issues/6059) on which this PR issue is blocked by.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-entity-store-performance-from-pr --> Click to trigger **kibana-entity-store-performance-from-pr** for this PR!

stefnestor commented on x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_request_diagnostics_modal/index.tsx:
```suggestion
            defaultMessage="Include Events Logs (might contain sensitive information)"
```
`sensitive` not `sensible`  🙂
